### PR TITLE
new branch of MS Sculpt mobile keyboard, keymaps and matching userspace

### PR DESCRIPTION
**Pull request by milestogo**
update ms_sculpt_mobile handwired keyboard to current chibios, new directory structure for multiple boards, and current userspace. 

## Description
This is a restart of https://github.com/qmk/qmk_firmware/pull/9197 , which had drifted too far from master and my current keymaps.  This change splits the 3 possible MCUs into different subfolders. Made many updates to work with current ChibiOS. Updated  documentation for babblepaste library to md format. 

If needed, I can split the userspace and my keymaps out from the keyboard update. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist
- [X ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/ms_sculpt_mobile/8x18_arm/config.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+pragma once
```
Comment (drashna) [inline keyboards/handwired/ms_sculpt_mobile/8x18_arm/config.h:18]:
```suggestion
#pragma once

```

File: keyboards/handwired/ms_sculpt_mobile/8x18_arm/config.h [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID     0xFEED
+#define PRODUCT_ID    0x01ED
+#define DEVICE_VER      0x0001
+
+#define MANUFACTURER "QMK"
+#define PRODUCT "8x18"
```
Comment (drashna) [inline keyboards/handwired/ms_sculpt_mobile/8x18_arm/config.h:27]:
```suggestion
#define MANUFACTURER QMK
#define PRODUCT 8x18
```

File: keyboards/handwired/ms_sculpt_mobile/keymaps/matrix/config.h [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#pragma once
```
Comment (drashna) [inline keyboards/handwired/ms_sculpt_mobile/keymaps/matrix/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/handwired/ms_sculpt_mobile/keymaps/matrix/keymap.c [hunk 4]
```diff
@@ -0,0 +1,30 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/handwired/ms_sculpt_mobile/keymaps/matrix/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/handwired/ms_sculpt_mobile/keymaps/matrix/keymap.c [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT_FULLMATRIX_8x18( \
```
Comment (drashna) [inline keyboards/handwired/ms_sculpt_mobile/keymaps/matrix/keymap.c:4]:
```suggestion
  [0] = LAYOUT_fullmatrix_8x18( \
```

File: keyboards/handwired/ms_sculpt_mobile/ms_sculpt_mobile.h [hunk 6]
```diff
@@ -1,30 +1,70 @@
...
+        {k2A, k2B, k2C, k2D, k2E, k2F, XXX, k2H, k2I, XXX, XXX, XXX, k2M, k2N, XXX, XXX, XXX, XXX},\
+        {k3A, XXX, k3C, k3D, k3E, k3F, k3G, k3H, k3I, XXX, XXX, XXX, XXX, XXX, XXX, XXX, XXX, k3R},\
+        {k4A, k4B, k4C, k4D, k4E, k4F, k4G, k4H, k4I, XXX, XXX, XXX, XXX, XXX, k4O, XXX, XXX, XXX},\
+        {k5A, k5B, k5C, k5D, k5E, k5F, k5G, k5H, XXX, XXX, XXX, XXX, XXX, XXX, XXX, XXX, XXX, XXX},\
+        {k6A, k6B, k6C, k6D, k6E, k6F, k6G, k6H, k6I, k6J, k6K, k6L, k6M, k6N, XXX, XXX, XXX, XXX},\
+        {k7A, k7B, k7C, k7D, k7E, k7F, k7G, k7H, k7I, XXX, k7K, k7L, k7M, XXX, XXX, k7P, XXX, XXX},\
+}
+
+// dummy keymap for finding all locations. Use the matrix keymap & hid listen to find keys
+#define LAYOUT_FULLMATRIX_8x18( \
```
Comment (drashna) [inline keyboards/handwired/ms_sculpt_mobile/ms_sculpt_mobile.h:67]:
```suggestion
#define LAYOUT_fullmatrix_8x18( \
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/handwired/ms_sculpt_mobile/astar/rules.mk [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/handwired/ms_sculpt_mobile/astar/rules.mk:17]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

File: keyboards/handwired/ms_sculpt_mobile/teensypp/rules.mk [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/handwired/ms_sculpt_mobile/teensypp/rules.mk:19]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
```

File: keyboards/handwired/ms_sculpt_mobile/teensypp/rules.mk [hunk 9]
```diff
@@ -0,0 +1,20 @@
+MCU = at90usb1286
+
+# Bootloader Selection
+BOOTLOADER = halfkay
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```
Comment (fauxpark) [inline keyboards/handwired/ms_sculpt_mobile/teensypp/rules.mk:9]:
```suggestion
BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
```

File: keyboards/handwired/ms_sculpt_mobile/teensypp/config.h [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+ *
+ * Magic keys are hotkey commands that allow control over firmware functions of
+ * the keyboard. They are best used in combination with the HID Listen program,
+ * found here: https://www.pjrc.com/teensy/hid_listen.html
+ *
+ * The options below allow the magic key functionality to be changed. This is
+ * useful if your keyboard/keypad is missing keys and you want magic key support.
+ *
+ */
+
```
Comment (fauxpark) [inline keyboards/handwired/ms_sculpt_mobile/teensypp/config.h:60]:
```suggestion
```

File: keyboards/handwired/ms_sculpt_mobile/keymaps/milestogo/rules.mk [hunk 11]
```diff
@@ -2,16 +2,24 @@
 #   change to "no" to disable the options, or define them in the Makefile in
```
Comment (fauxpark) [inline keyboards/handwired/ms_sculpt_mobile/keymaps/milestogo/rules.mk:2]: You probably don't need most of this in a keymap rules.mk.

File: keyboards/handwired/ms_sculpt_mobile/keymaps/matrix/rules.mk [hunk 12]
```diff
@@ -0,0 +1,17 @@
+# Build Options
```
Comment (fauxpark) [inline keyboards/handwired/ms_sculpt_mobile/keymaps/matrix/rules.mk:1]: As above.

File: keyboards/handwired/ms_sculpt_mobile/keymaps/matrix/keymap.c [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+		    
+  KC_A,  KC_B,   KC_C,   KC_D,   KC_E,   KC_F,   KC_G,   KC_H,   KC_I,   KC_J,   KC_K,   KC_L,   KC_M,   KC_N,   KC_O,   KC_P,      KC_Q,   KC_R,\
+  KC_A,  KC_B,   KC_C,   KC_D,   KC_E,   KC_F,   KC_G,   KC_H,   KC_I,   KC_J,   KC_K,   KC_L,   KC_M,   KC_N,   KC_O,   KC_P,      KC_Q,   KC_R,\
+  KC_A,  KC_B,   KC_C,   KC_D,   KC_E,   KC_F,   KC_G,   KC_H,   KC_I,   KC_J,   KC_K,   KC_L,   KC_M,   KC_N,   KC_O,   KC_P,      KC_Q,   KC_R,\
+  KC_A,  KC_B,   KC_C,   KC_D,   KC_E,   KC_F,   KC_G,   KC_H,   KC_I,   KC_J,   KC_K,   KC_L,   KC_M,   KC_N,   KC_O,   KC_P,      KC_Q,   KC_R,\
+  KC_A,  KC_B,   KC_C,   KC_D,   KC_E,   KC_F,   KC_G,   KC_H,   KC_I,   KC_J,   KC_K,   KC_L,   KC_M,   KC_N,   KC_O,   KC_P,      KC_Q,   KC_R,\
+  KC_A,  KC_B,   KC_C,   KC_D,   KC_E,   KC_F,   KC_G,   KC_H,   KC_I,   KC_J,   KC_K,   KC_L,   KC_M,   KC_N,   KC_O,   KC_P,      KC_Q,   KC_R,\
+  KC_A,  KC_B,   KC_C,   KC_D,   KC_E,   KC_F,   KC_G,   KC_H,   KC_I,   KC_J,   KC_K,   KC_L,   KC_M,   KC_N,   KC_O,   KC_P,      KC_Q,   KC_R,\
+  KC_A,  KC_B,   KC_C,   KC_D,   KC_E,   KC_F,   KC_G,   KC_H,   KC_I,   KC_J,   KC_K,   KC_L,   KC_M,   KC_N,   KC_O,   KC_P,      KC_Q,   KC_R\
+  )\
```
Comment (fauxpark) [inline keyboards/handwired/ms_sculpt_mobile/keymaps/matrix/keymap.c:31]:
```suggestion
  [0] = LAYOUT_fullmatrix_8x18(
  KC_A,  KC_B,   KC_C,   KC_D,   KC_E,   KC_F,   KC_G,   KC_H,   KC_I,   KC_J,   KC_K,   KC_L,   KC_M,   KC_N,   KC_O,   KC_P,      KC_Q,   KC_R,
  KC_A,  KC_B,   KC_C,   KC_D,   KC_E,   KC_F,   KC_G,   KC_H,   KC_I,   KC_J,   KC_K,   KC_L,   KC_M,   KC_N,   KC_O,   KC_P,      KC_Q,   KC_R,
  KC_A,  KC_B,   KC_C,   KC_D,   KC_E,   KC_F,   KC_G,   KC_H,   KC_I,   KC_J,   KC_K,   KC_L,   KC_M,   KC_N,   KC_O,   KC_P,      KC_Q,   KC_R,
  KC_A,  KC_B,   KC_C,   KC_D,   KC_E,   KC_F,   KC_G,   KC_H,   KC_I,   KC_J,   KC_K,   KC_L,   KC_M,   KC_N,   KC_O,   KC_P,      KC_Q,   KC_R,
  KC_A,  KC_B,   KC_C,   KC_D,   KC_E,   KC_F,   KC_G,   KC_H,   KC_I,   KC_J,   KC_K,   KC_L,   KC_M,   KC_N,   KC_O,   KC_P,      KC_Q,   KC_R,
  KC_A,  KC_B,   KC_C,   KC_D,   KC_E,   KC_F,   KC_G,   KC_H,   KC_I,   KC_J,   KC_K,   KC_L,   KC_M,   KC_N,   KC_O,   KC_P,      KC_Q,   KC_R,
  KC_A,  KC_B,   KC_C,   KC_D,   KC_E,   KC_F,   KC_G,   KC_H,   KC_I,   KC_J,   KC_K,   KC_L,   KC_M,   KC_N,   KC_O,   KC_P,      KC_Q,   KC_R,
  KC_A,  KC_B,   KC_C,   KC_D,   KC_E,   KC_F,   KC_G,   KC_H,   KC_I,   KC_J,   KC_K,   KC_L,   KC_M,   KC_N,   KC_O,   KC_P,      KC_Q,   KC_R
  )
```

File: keyboards/handwired/ms_sculpt_mobile/keymaps/matrix/config.h [hunk 14]
```diff
@@ -0,0 +1,20 @@
+/*
```
Comment (fauxpark) [inline keyboards/handwired/ms_sculpt_mobile/keymaps/matrix/config.h:1]: This file can be deleted.

File: keyboards/handwired/ms_sculpt_mobile/keymaps/default/keymap.c [hunk 15]
```diff
@@ -17,12 +17,12 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
...
-    KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, RSFT(KC_1), KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT  \
-  )
+[0] = LAYOUT_mobile_XUW( \
+	KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,     KC_F12,  KC_VOLD, KC_VOLU, KC_MUTE, \
+	KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS,    KC_EQL,           KC_BSPC, KC_DEL,  \
+	KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC,    KC_RBRC,          KC_BSLS, \
+	KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,                      KC_ENT,  KC_PGUP, \
+	KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,                      KC_RSFT, KC_UP,   KC_PGDN, \
+	KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, RSFT(KC_1), KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT  \
```
Comment (fauxpark) [inline keyboards/handwired/ms_sculpt_mobile/keymaps/default/keymap.c:26]:
```suggestion
[0] = LAYOUT_mobile_XUW(
	KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,     KC_F12,  KC_VOLD, KC_VOLU, KC_MUTE,
	KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS,    KC_EQL,           KC_BSPC, KC_DEL,
	KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC,    KC_RBRC,          KC_BSLS,
	KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,                      KC_ENT,  KC_PGUP,
	KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,                      KC_RSFT, KC_UP,   KC_PGDN,
	KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, RSFT(KC_1), KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
```

File: keyboards/handwired/ms_sculpt_mobile/8x18_arm/rules.mk [hunk 16]
```diff
@@ -0,0 +1,18 @@
...
+
+## Chibi options that should no longer be needed. 
+##-DCORTEX_VTOR_INIT=0x2000
+# Options to pass to dfu-util when flashing
+##DFU_ARGS = -d 1EAF:0003 -a 2 -R
+##DFU_SUFFIX_ARGS  ?= -v 1EAF -p 0003
+
+# avoid warning about LTO & ChibiOS, even though it mostly works
+#LTO_ENABLE = no
+
```
Comment (fauxpark) [inline keyboards/handwired/ms_sculpt_mobile/8x18_arm/rules.mk:18]:
```suggestion
# MCU name
MCU = STM32F103

# Bootloader selection
BOOTLOADER = stm32duino

WS2812_DRIVER = bitbang

# Enter lower-power sleep mode when on the ChibiOS idle thread
OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE
```

File: keyboards/handwired/ms_sculpt_mobile/8x18_arm/8x18_arm_v1.txt [hunk 17]
```diff
@@ -0,0 +1,33 @@
+
```
Comment (fauxpark) [inline keyboards/handwired/ms_sculpt_mobile/8x18_arm/8x18_arm_v1.txt:1]: Please delete this file.

File: keyboards/handwired/ms_sculpt_mobile/8x18_arm/8x18_arm_v0.txt [hunk 18]
```diff
@@ -0,0 +1,45 @@
+ribbon pin,  teensy chip pin,	bluepill,	teensy2++
```
Comment (fauxpark) [inline keyboards/handwired/ms_sculpt_mobile/8x18_arm/8x18_arm_v0.txt:1]: Please delete this file.

File: keyboards/handwired/ms_sculpt_mobile/8x18_arm/8x18_arm.c [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+
+#include "8x18_arm.h"
+
+void keyboard_pre_init_kb(void) {
+    // init the CAPS LOCK LED pin as an output
+    setPinOutput(STM32LED);
+
+    // disable JTAG  to enable PB3 and PB4
+    AFIO->MAPR |= AFIO_MAPR_SWJ_CFG_JTAGDISABLE;
+    AFIO->MAPR |= AFIO_MAPR_SWJ_CFG_DISABLE;
```
Comment (fauxpark) [inline keyboards/handwired/ms_sculpt_mobile/8x18_arm/8x18_arm.c:27]: This should probably be done in a `boardInit()` override.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/ms_sculpt_mobile/keymaps/matrix/rules.mk (hunk 12 above)
Comment (milestogo) [reply keyboards/handwired/ms_sculpt_mobile/keymaps/matrix/rules.mk:1]: minimized

File: keyboards/handwired/ms_sculpt_mobile/ms_sculpt_mobile.h [hunk 20]
```diff
@@ -1,43 +1,90 @@
...
-    { k00, k01, k02, k03, k04, k05, XXX, XXX, XXX, k09, XXX, XXX, XXX, XXX, XXX, XXX, XXX, k0H }, \
-    { k10, k11, k12, XXX, XXX, XXX, XXX, XXX, XXX, XXX, k1A, k1B, k1C, XXX, k1E, k1F, k1G, XXX }, \
-    { k20, k21, k22, k23, k24, XXX, k26, XXX, XXX, XXX, XXX, k2B, XXX, XXX, XXX, k2F, XXX, XXX }, \
-    { k30, k31, k32, k33, XXX, XXX, XXX, XXX, XXX, XXX, k3A, XXX, XXX, XXX, k3E, XXX, XXX, XXX }, \
-    { k40, k41, k42, k43, XXX, XXX, XXX, XXX, XXX, XXX, XXX, XXX, XXX, k4D, XXX, XXX, XXX, XXX }, \
-    { k50, k51, k52, k53, k54, k55, k56, k57, k58, k59, k5A, k5B, k5C, k5D, k5E, k5F, k5G, XXX }, \
-    { k60, k61, k62, k63, k64, k65, k66, k67, k68, k69, XXX, XXX, XXX, k6D, k6E, k6F, k6G, k6H }, \
-    { k70, k71, k72, k73, k74, k75, XXX, k77, k78, k79, k7A, k7B, k7C, XXX, k7E, k7F, k7G, XXX } \
+//US layout, with vol-, vol+, mute in upper right, white Fn keys
+#define LAYOUT_mobile_XUA( \
```
Comment (noroadsleft) [inline keyboards/handwired/ms_sculpt_mobile/ms_sculpt_mobile.h:30]: Layout macro named should be fully lowercase except for the `LAYOUT` prefix.

* [Keyboard Guidelines - `<keyboard_name.h>` - QMK Firmware Documentation](https://docs.qmk.fm/#/hardware_keyboard_guidelines?id=keyboard-folder-structure)

```suggestion
#define LAYOUT_mobile_xua( \
```

File: keyboards/handwired/ms_sculpt_mobile/ms_sculpt_mobile.h [hunk 21]
```diff
@@ -1,43 +1,90 @@
...
+        {k3A, k3B, k3C, k3D, XXX, XXX, XXX, XXX, XXX, XXX, k3K, XXX, XXX, XXX, k3O, XXX, XXX, XXX},\
+        {k4A, k4B, k4C, k4D, XXX, XXX, XXX, XXX, XXX, XXX, XXX, XXX, XXX, k4N, XXX, XXX, XXX, XXX},\
+        {k5A, k5B, k5C, k5D, k5E, k5F, k5G, k5H, k5I, k5J, k5K, k5L, k5M, k5N, k5O, k5P, k5Q, XXX},\
+        {k6A, k6B, k6C, k6D, k6E, k6F, k6G, k6H, k6I, k6J, XXX, XXX, XXX, k6N, k6O, k6P, k6Q, k6R},\
+        {k7A, k7B, k7C, k7D, k7E, k7F, XXX, k7H, k7I, k7J, k7K, k7L, k7M, XXX, k7O, k7P, k7Q, XXX},\
+}
+
+
+//NA windows? blue overlay on FN keys, PrtScn, HOme, End in upper right. 
+#define LAYOUT_mobile_XUW( \
```
Comment (noroadsleft) [inline keyboards/handwired/ms_sculpt_mobile/ms_sculpt_mobile.h:51]:
```suggestion
#define LAYOUT_mobile_xuw( \
```

File: keyboards/handwired/ms_sculpt_mobile/teensypp/teensypp.c [hunk 22]
```diff
@@ -0,0 +1 @@
+#include "teensypp.h"
```
Comment (noroadsleft) [inline keyboards/handwired/ms_sculpt_mobile/teensypp/teensypp.c:1]: This file needs a GPLv2-compatible license header.

* [PR Checklist - Example GPLv2 Header - QMK Firmware Documentation](https://docs.qmk.fm/#/pr_checklist?id=example-gplv2-header)

File: keyboards/handwired/ms_sculpt_mobile/teensypp/teensypp.h [hunk 23]
```diff
@@ -0,0 +1,2 @@
+#pragma once
```
Comment (noroadsleft) [inline keyboards/handwired/ms_sculpt_mobile/teensypp/teensypp.h:1]: This file needs a GPLv2-compatible license header.

* [PR Checklist - Example GPLv2 Header - QMK Firmware Documentation](https://docs.qmk.fm/#/pr_checklist?id=example-gplv2-header)

File: keyboards/handwired/ms_sculpt_mobile/keymaps/milestogo/keymap.c [hunk 24]
```diff
@@ -1,261 +1,349 @@
...
-SYM,
-MOV,
-NUM,
-TRAN
-};
+#include "milestogo.h"
+#include "virtser.h"
+#include <print.h>
 
+#define LAYOUT_local LAYOUT_mobile_XUA
```
Comment (noroadsleft) [inline keyboards/handwired/ms_sculpt_mobile/keymaps/milestogo/keymap.c:24]:
```suggestion
#define LAYOUT_local LAYOUT_mobile_xua
```

File: keyboards/handwired/ms_sculpt_mobile/keymaps/default/rules.mk [hunk 25]
```diff
@@ -1,5 +1,5 @@
 BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
-MOUSEKEY_ENABLE = yes       # Mouse keys(+4700)
+MOUSEKEY_ENABLE = no       # Mouse keys(+4700)
```
Comment (noroadsleft) [inline keyboards/handwired/ms_sculpt_mobile/keymaps/default/rules.mk:2]:
```suggestion
MOUSEKEY_ENABLE = no        # Mouse keys
```

File: keyboards/handwired/ms_sculpt_mobile/keymaps/default/keymap.c [hunk 26]
```diff
@@ -17,12 +17,12 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
...
 
-  [0] = LAYOUT( \
-    KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,     KC_F12,  KC_VOLD, KC_VOLU, KC_MUTE, \
-    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS,    KC_EQL,           KC_BSPC, KC_DEL,  \
-    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC,    KC_RBRC,          KC_BSLS, \
-    KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,                      KC_ENT,  KC_PGUP, \
-    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,                      KC_RSFT, KC_UP,   KC_PGDN, \
-    KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, RSFT(KC_1), KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT  \
-  )
+[0] = LAYOUT_mobile_XUW(
```
Comment (noroadsleft) [inline keyboards/handwired/ms_sculpt_mobile/keymaps/default/keymap.c:20]: This file needs a GPLv2-compatible license header.

* [PR Checklist - Example GPLv2 Header - QMK Firmware Documentation](https://docs.qmk.fm/#/pr_checklist?id=example-gplv2-header)

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/handwired/ms_sculpt_mobile/teensypp/teensypp.c (hunk 22 above)
Comment (milestogo) [reply keyboards/handwired/ms_sculpt_mobile/teensypp/teensypp.c:1]: I thought at one point the style was to not require license for header files of less than 4 lines. Adding.

**Review by noroadsleft — APPROVED** [review]
Looks good to me.

**Comment by fauxpark** [comment]
Would prefer #16038 to go in first.

**Pull request closed without merging**